go/vt/vtgate/planbuilder/update.go

```diff
@@ -34,19 +34,17 @@ func buildUpdatePlan(stmt sqlparser.Statement, vschema ContextVSchema) (engine.P
 		return nil, err
 	}
 	eupd := &engine.Update{
-		DML:                 *dml,
-		ChangedVindexValues: make(map[string]engine.VindexValues),
+		DML: *dml,
 	}
 
 	if dml.Opcode == engine.Unsharded {
 		return eupd, nil
 	}
 
-	if eupd.ChangedVindexValues, err = buildChangedVindexesValues(upd, eupd.Table.ColumnVindexes); err != nil {
+	if eupd.ChangedVindexValues, eupd.OwnedVindexQuery, err = buildChangedVindexesValues(upd, eupd.Table, ksidCol); err != nil {
```

Comment:
nit: feels unclean to assign to `eupd` without checking for errors. I know the old code was doing it